services/errors.js

```diff
@@ -1,34 +1,60 @@
 'use strict';
 
-class NotFound extends Error {
-  constructor(prettyMessage = 'not found') {
+class ShieldsRuntimeError extends Error {
+
+  get name() { return 'ShieldsRuntimeError'; }
+  get defaultPrettyMessage() { throw new Error('Must implement abstract method'); }
+
+  constructor(props, message) {
+    props = props || {};
+    super(message);
+    this.prettyMessage = props.prettyMessage || this.defaultPrettyMessage;
+    if (props.underlyingError) {
+      this.stack = props.underlyingError.stack;
+    }
+  }
+}
+
+class NotFound extends ShieldsRuntimeError {
+
+  get name() { return 'NotFound'; }
+  get defaultPrettyMessage() { return 'not found'; }
+
+  constructor(props) {
+    props = props || {};
+    const prettyMessage = props.prettyMessage || 'not found';
```

Comment:
should this line _(& line below)_ be using `this.defaultPrettyMessage` in place of `'not found'`

Reply:
That would be ideal, but there's a catch 22:

If we try to reference `this` before `super()`, `error  'this' is not allowed before 'super()'` is thrown (which makes sense), but we need to assemble `message` before we can call `super()`.

Maybe an alternative would be to declare the default messages as constants outside the classes so we can say

```js
const defaultNotFoundMessage = 'not found';

class NotFound extends ShieldsRuntimeError {

  get name() { return 'NotFound'; }
  get defaultPrettyMessage() { return defaultNotFoundMessage; }

  constructor(props = {}) {
    const prettyMessage = props.prettyMessage || defaultNotFoundMessage;
    //...
  }
}
```

do you think that is better?

Reply:
Ah yeah that makes sense, sounds good to me :+1: 
You can keep it how it is or change it, whichever makes the most sense for you.